deployment-server/src/Deploy/Worker.hs

```diff
@@ -9,77 +9,83 @@ module Deploy.Worker where
     withSystemTempDirectory "deployment" $ \tempDir -> do
-            let plutusDir = tempDir <> "/plutus"
-                nixopsDir = plutusDir <> "/deployment/nixops"
-                extraIncludesString = fmap (\s -> "-I" <> s) include
-                args =
-                      extraIncludesString <>
-                      [ "-s"
-                      , stateFile
-                      , "-d"
-                      , (Text.unpack . unpack) deploymentName
-                      ]
-                ref = Text.unpack . whPullReqTargetRef . whPullReqBase . evPullReqPayload $ event
+            let ref = Text.unpack . whPullReqTargetRef . whPullReqBase . evPullReqPayload $ event
+                auth = mkToken githubToken
             putStrLn $ "Deploy origin/" <> ref
-            result <- runExceptT $ do
-              void $ runIn tempDir "git" ["clone", "https://github.com/input-output-hk/plutus.git"]
-              void $ runIn plutusDir "git" ["checkout", "origin/" <> ref]
-              (_, gitHeadStdout, _) <- runIn plutusDir "git" ["rev-parse", "HEAD"]
-              let gitHead = Text.pack . rstrip . BS.unpack $ gitHeadStdout
-              jsonFiles <- filter isJson <$> liftIO (listDirectory configDir)
-              -- we copy from the configDir in case someone has tried to be nasty by providing
-              -- configDir like "/etc/passwd /proper/config/dir", this means configDir must be
-              -- a proper directory otherwise runIn will fail
-              void $ runIn configDir "cp" $ ["-p"] <> jsonFiles <> [nixopsDir]
-              liftIO $ putStrLn $ "deploy " <> (Text.unpack . unpack) deploymentName
-              void $ runIn nixopsDir "nixops" $ ["modify", "./default.nix", "./network.nix"] <> args
-              -- We do a rolling deploy, if availability zone A fails to deploy or health check fails then we will
-              -- investigate and roll forward, rather than rolling back
-              void $ runIn nixopsDir "nixops" $ ["deploy", "--include", "playgroundA", "--include", "marlowePlaygroundA"] <> args
-              checkHealth nixopsDir A environment
-              void $ runIn nixopsDir "nixops" $ ["deploy", "--include", "playgroundB", "--include", "marlowePlaygroundB"] <> args
-              checkHealth nixopsDir B environment
-              pure gitHead
-            putStrLn $ "finished deployment with result: " <> show result
-            alert slackConfig slackChannel (showResult result)
+            deploymentResult <- runExceptT $ do
+                  deployment <- ExceptT $ createGithubDeployment ref environment auth
+                  deploymentResult <- runExceptT $ runDeployment tempDir ref
+                  ExceptT $ completeGithubDeployment auth deployment deploymentResult
+            case deploymentResult of
+                    Left err -> putStrLn $ "deployment failed with: " <> show err
+                    Right _  -> putStrLn "deployment finished without errors"
```

Comment:
Since you have `gitHead` here as the `Right` value, it seems a shame not to stick it in the result message...


Also, I'd be very tempted to lift the put to `putStrLn $ case ...`

Reply:
ok done, there was a naming conflict, this was not actually the git head but I have changed that so that it is a nested `Either`, see latest commit.